packages/velo-external-db-core/lib/service/data.spec.js

```diff
@@ -4,24 +4,30 @@ const driver = require('../../test/drivers/data_provider_test_support')
 const schema = require('../../test/drivers/schema_information_test_support')
 const filterTransformer = require ('../../test/drivers/filter_transformer_test_support')
 const aggregationTransformer = require('../../test/drivers/aggregation_transformer_test_support')
+const queryValidator = require('../../test/drivers/query_validator_test_support')
 const { AdapterOperators } = require('velo-external-db-commons')
 const Chance = require('chance')
 const chance = new Chance()
 
 describe('Data Service', () => {
 
-    // eslint-disable-next-line jest/no-disabled-tests
-    test.skip('delegate request to data provider and translate data to velo format', async() => {
+    test('delegate request to data provider and translate data to velo format', async() => {
         driver.givenListResult(ctx.entities, ctx.collectionName, ctx.filter, ctx.sort, ctx.skip, ctx.limit)
-        filterTransformer.stubIgnoreTransform(ctx.filter)
+        schema.givenDefaultSchemaFor(ctx.collectionName)
+        filterTransformer.givenTransformResult(ctx.filter)
+
+        const transformedFilter = filterTransformer.transform(ctx.filter)
+        const fields = await schema.schemaInformation.schemaFieldsFor(ctx.collectionName)
+        
+        queryValidator.expectValidFilter(fields, transformedFilter)
 
         const actual = await env.dataService.find(ctx.collectionName, ctx.filter, ctx.sort, ctx.skip, ctx.limit)
         expect( actual ).toEqual({ items: ctx.entities, totalCount: ctx.entities.length })
```

Comment:
on the way we need to fix this line to be expect().resolves(....) vs what's written here right now.